contracts/swindle.sol

```diff
@@ -1,86 +1,101 @@
 pragma solidity ^0.4.0;
 
-import "./abstracts/CourtroomAbstract.sol";
+import "./abstracts/swindleabstract.sol";
 import "./sampletoken.sol";
 import "./abstracts/trialrulesabstract.sol";
+import "./abstracts/swearabstract.sol";
 
 
-contract SwearGame is SwearGameAbstract {
+contract Swindle is SwindleAbstract {
 
-    uint256 public deposit;
-    uint  public playerCount;
-    SampleToken public token;
     TrialRulesAbstract public trialRules;
+    SwearAbstract public swear;
 
     struct Case {
         address plaintiff;
         bytes32 serviceId;
         uint8 status;
         uint8 valid;
     }
-
-    //id map to _Case
+    //id map to Case
     mapping(bytes32 => Case)  OpenCases;
-    mapping(address => bool) public players;
     mapping(address => bytes32[]) public ids;
 
-    /// @notice SwearGame - Swear game constructor this function is called along with
+    /// @notice Swindle - Swindle constructor this function is called along with
     /// the contract deployment time.
-    ///
-    /// @param _token address of the token contract
+    /// @param _swear - address of the swear contract
     /// @param _trialRules - address of the trial specific rules contract
     /// @return WitnessAbstract - return a witness contract instance
-    function SwearGame(address _token, address _trialRules) {
-
-        token = SampleToken(_token);
+    function Swindle(address _swear,address _trialRules) {
+        swear = SwearAbstract(_swear);
+        require(swear.setSwindleContractAddress(address(this)));
```

Comment:
Is it not better to create a swear instance from the swindle?

Reply:
Is that due to simplification of the contract suite deployment or there is another reason?
 
We can do that though in that case external access the swear contact will be via the swindle contract or by calling the swindle to get swear address and then call it.